### PR TITLE
Add gRPC greeter sample application

### PR DESCRIPTION
**Pull request by halvards**
This sample application consists of a gRPC server and client.

It is a adapted from the gRPC-Go helloworld example:
https://github.com/grpc/grpc-go/tree/master/examples/helloworld

The main changes from the original version are:

- the server returns the hostname where it runs in a gRPC header; and

- the client prints the hostname header from the gRPC response.

This is useful for testing request routing and load balancing
functionality.

When running in a Kubernetes cluster, the hostname is the name of the
Kubernetes Pod that served the request.

### CODE REVIEW

File: sample-apps/grpc-greeter-go/greeter-istio.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: greeter
+spec:
+  selector:
+    istio: ilbgateway
```
Comment (askmeegs) [inline sample-apps/grpc-greeter-go/greeter-istio.yaml]: This sample app will live in the generic `sample-apps/` directory, not coupled with an ILB tutorial. I wonder if having ILBGateway as default here is a good idea, vs. `ingressgateway`.

File: sample-apps/grpc-greeter-go/greeter-istio.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# Copyright 2019 Google LLC
```
Comment (askmeegs) [inline sample-apps/grpc-greeter-go/greeter-istio.yaml]: Move both YAML manifests into `manifests/` or `k8s/` directory? Thanks.

File: sample-apps/grpc-greeter-go/greeter-k8s.template.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    matchLabels:
+      app: greeter
+  template:
+    metadata:
+      labels:
+        app: greeter
+    spec:
+      containers:
+      - name: greeter
+        image: gcr.io/$GOOGLE_CLOUD_PROJECT/grpc-greeter-go-server
```
Comment (askmeegs) [inline sample-apps/grpc-greeter-go/greeter-k8s.template.yaml]: Why only a deployment for the server, not the client?

File: sample-apps/grpc-greeter-go/greeter-istio.yaml (hunk 1 above)
Comment (halvards) [reply sample-apps/grpc-greeter-go/greeter-istio.yaml]: Good point. I've added two gateway manifests, one for Ingress Gateway and one for ILB Gateway, so users can choose which one to apply.

File: sample-apps/grpc-greeter-go/greeter-istio.yaml (hunk 2 above)
Comment (halvards) [reply sample-apps/grpc-greeter-go/greeter-istio.yaml]: Moved to manifests/

File: sample-apps/grpc-greeter-go/greeter-k8s.template.yaml (hunk 3 above)
Comment (halvards) [reply sample-apps/grpc-greeter-go/greeter-k8s.template.yaml]: The client isn't a long-running process, it just makes some requests to the server and then exits. We could create a k8s job for it, if you think that'll be useful?

**Comment by halvards** [comment]
Thanks for the review @askmeegs ! I added a commit with updates. Please let me know if you want me to squash them.

File: sample-apps/grpc-greeter-go/greeter-k8s.template.yaml (hunk 3 above)
Comment (askmeegs) [reply sample-apps/grpc-greeter-go/greeter-k8s.template.yaml]: Ok! No, I think just the one deployment is fine, then.

**Pull request merged**